MonoGame.Framework.Content.Pipeline/OpenAssetImporter.cs

```diff
@@ -366,6 +384,82 @@ private void ImportMaterials()
+        /// </summary>
+        private void ImportAssImpMaterials()
+        {
+            _materials = new List<MaterialContent>();
+            foreach (var aiMaterial in _scene.Materials)
+            {
+                // TODO: Should we create a special AssetImporterMaterial?
+
+                var material = new MaterialContent
+                {
+                    Name = aiMaterial.Name,
+                    Identity = _identity,
+                };
+
+                var slots = aiMaterial.GetAllMaterialTextures();
+                foreach (var tex in slots)
+                {
+                    var name = tex.TextureType.ToString();
+                    if (tex.TextureIndex != 0)
+                        name += tex.TextureIndex;
+                    material.Textures.Add(name, ImportTextureContentRef(tex));
+                }
+
+                if (aiMaterial.HasBlendMode)
+                    material.OpaqueData.Add("BlendMode", aiMaterial.BlendMode.ToString());
+                if (aiMaterial.HasBumpScaling)
+                    material.OpaqueData.Add("BumpScaling", aiMaterial.BumpScaling);
+                if (aiMaterial.HasColorAmbient)
+                    material.OpaqueData.Add("ColorAmbient", ToXna(aiMaterial.ColorAmbient));
+                if (aiMaterial.HasColorDiffuse)
+                    material.OpaqueData.Add("ColorDiffuse", ToXna(aiMaterial.ColorDiffuse));
+                if (aiMaterial.HasColorEmissive)
+                    material.OpaqueData.Add("ColorEmissive", ToXna(aiMaterial.ColorEmissive));
+                if (aiMaterial.HasColorReflective)
+                    material.OpaqueData.Add("ColorReflective", ToXna(aiMaterial.ColorReflective));
+                if (aiMaterial.HasColorSpecular)
+                    material.OpaqueData.Add("ColorSpecular", ToXna(aiMaterial.ColorSpecular));
+                if (aiMaterial.HasColorTransparent)
+                    material.OpaqueData.Add("ColorTransparent", ToXna(aiMaterial.ColorTransparent));
+                if (aiMaterial.HasOpacity)
+                    material.OpaqueData.Add("Opacity", aiMaterial.Opacity);
+                if (aiMaterial.HasReflectivity)
+                    material.OpaqueData.Add("Reflectivity", aiMaterial.Reflectivity);
+                if (aiMaterial.HasShadingMode)
+                    material.OpaqueData.Add("ShadingMode", aiMaterial.ShadingMode.ToString());
+                if (aiMaterial.HasShininess)
+                    material.OpaqueData.Add("Shininess", aiMaterial.Shininess);
+                if (aiMaterial.HasShininessStrength)
+                    material.OpaqueData.Add("ShininessStrength", aiMaterial.ShininessStrength);
+                if (aiMaterial.HasTwoSided)
+                    material.OpaqueData.Add("TwoSided", aiMaterial.IsTwoSided);
+                if (aiMaterial.HasWireFrame)
+                    material.OpaqueData.Add("WireFrame", aiMaterial.IsWireFrameEnabled);
```

Comment:
We're also using AssImp's naming convention for material properties... `ColorDiffuse` vs `DiffuseColor`.

Reply:
Still thinking about this...   AssImp's convention comes from their material property keys like `AI_MATKEY_COLOR_DIFFUSE`.  

Thinking now that we're better off with a more readable naming than a C++ naming convention designed for editor autocompletion..

Reply:
AssImp's material docs:

 http://assimp.sourceforge.net/lib_html/materials.html
 http://assimp.sourceforge.net/lib_html/material_8h.html

Reply:
IMO we don't have to stick with assimp naming. 99.9% of the people using this won't be familiar with the assimp API anyway. We just have to make sure to document the mapping in case people want to reference the assimp docs.

Reply:
Yeah i think i will make the naming more XNA like and spend some time writing a doc page on the details of the `OpenAssetImporter`.